dissect/target/plugins/apps/webserver/apache.py

```diff
@@ -77,83 +196,165 @@ def get_log_paths(self) -> list[Path]:
+
+                yield WebserverErrorLogRecord(
+                    ts=ts,
+                    pid=log.get("pid"),
+                    remote_ip=remote_ip,
+                    module=log["module"],
+                    level=log["level"],
+                    error_source=error_source,
+                    error_code=error_code,
+                    message=log["message"],
+                    source=path,
+                    _target=self.target,
+                )
+
+            except Exception as e:
+                self.target.log.warning("An error occured parsing Apache log file %s: %s", path, str(e))
+                self.target.log.debug("", exc_info=e)
+
+    def _iterate_log_lines(self, paths: list[Path]) -> Iterator[tuple[str, Path]]:
+        """Iterate through a list of paths and yield tuples of loglines and the path of the file where they're from."""
+        for path in paths:
             try:
                 path = path.resolve(strict=True)
                 for line in open_decompress(path, "rt"):
                     line = line.strip()
                     if not line:
                         continue
+                    yield line, path
+            except FileNotFoundError:
+                self.target.log.warning("Apache log file configured but could not be found (dead symlink?): %s", path)
 
-                    fmt = infer_log_format(line)
-                    if not fmt:
-                        self.target.log.warning(
-                            "Apache log format could not be inferred for log line: %s (%s)", line, path
-                        )
-                        continue
+    @staticmethod
+    def infer_access_log_format(line: str) -> Optional[LogFormat]:
+        """Attempt to infer what standard LogFormat is used. Returns None if no known format can be inferred.
 
-                    match = fmt.value.match(line)
-                    if not match:
-                        self.target.log.warning(
-                            "Could not match Apache log format %s for log line: %s (%s)", fmt, line, path
-                        )
-                        continue
+        Three default log type examples from Apache (note that the ipv4 could also be ipv6)::
+            combined       = '1.2.3.4 - - [19/Dec/2022:17:25:12 +0100] "GET / HTTP/1.1" 304 247 "-" "Mozilla/5.0
+                            (Windows NT 10.0; Win64; x64); AppleWebKit/537.36 (KHTML, like Gecko) Chrome/108.0.0.0
+                            Safari/537.36"'
+            common         = '1.2.3.4 - - [19/Dec/2022:17:25:40 +0100] "GET / HTTP/1.1" 200 312'
+            vhost_combined = 'example.com:80 1.2.3.4 - - [19/Dec/2022:17:25:40 +0100] "GET / HTTP/1.1" 200 312 "-"
+                            "Mozilla/5.0 (Windows NT 10.0; Win64; x64); AppleWebKit/537.36 (KHTML, like Gecko)
+                            Chrome/108.0.0.0 Safari/537.36"'
```

Comment:
Did you check how this looks like in Sphinx?

Reply:
Made some small changes so it is properly formatted in Sphinx.